Merge IO browse and adv. search pages, refs #9141

jraddaoui opened the pull request:
- Include adv. search criteria and filters in a single expandable partial
- Add adv. search partial to IO browse page
- Mix facets and adv. search form and query in IO browse action
- Add browse options with print preview, export CSV and sort
- Add date sort option, ordering asc. over the events startDate
- Allow two kinds of date range query, inclusive or exact
- Hide repository facet and filter is multi-repository setting disabled
- Add changes to CSV export job including facet filtering
- Change routes adding topLod and showAdvanced params when needed
- Fix arElasticSearchPluginQuery, indexType not needed and facets required
- Leave search index action only for XHR requests from IO treeview search
- Improve AdvanceSearch JS prototype
- Add params like facets filters inside adv. search form to not loose them
- Add all selected facets and filters to header options remove filters
- Match facet and adv. filter url params to keep filter and facet in sync
- Improve part-of facet to show descendants and itself
- Include search-box query inside adv. search criteria
- Remove top level description filter from facets
- Don't show facets with only one item unless it's selected
- Small fixes for print preview
- Add delete option to criterions
- Disable/enable repository facet and filter based on top level descriptions
- Update top level autocomplete URL with repos select value and filter query
- Populate LOD filter with LOD facet terms
- Add settings (migrations and fixtures) to toggle some adv. search filters
- Move all page elements form to settings action, clean a lot of repeated code
- Remove sfThemePlugin, it was only used for the page elements form
- Improve unique data and conditions in language facet
- Stop using @ in facets

jraddaoui committed:
Merge IO browse and adv. search pages, refs #9141

- Include adv. search criteria and filters in a single expandable partial
- Add adv. search partial to IO browse page
- Mix facets and adv. search form and query in IO browse action
- Add browse options with print preview, export CSV and sort
- Add date sort option, ordering asc. over the events startDate
- Allow two kinds of date range query, inclusive or exact
- Hide repository facet and filter is multi-repository setting disabled
- Add changes to CSV export job including facet filtering
- Change routes adding topLod and showAdvanced params when needed
- Fix arElasticSearchPluginQuery, indexType not needed and facets required
- Leave search index action only for XHR requests from IO treeview search
- Improve AdvanceSearch JS prototype
- Add params like facets filters inside adv. search form to not loose them
- Add all selected facets and filters to header options remove filters
- Match facet and adv. filter url params to keep filter and facet in sync
- Improve part-of facet to show descentands and itself
- Include search-box query inside adv. search criteria
- Remove top level description filter from facets
- Don't show facets with only one item unless it's selected
- Small fixes for print preview

jraddaoui committed:
Add delete option to criterions

jraddaoui committed:
Fix collection header filter

jraddaoui committed:
Disable/enable repository facet and filter

- Based on if the top level descriptions autocomplete is populated
- Trigger change events when the autocomplete hidden input is updated

jraddaoui committed:
Update top level autocomplete URL with repo value

Obtained from the repos select and used to filter the autocomplete query

jraddaoui committed:
Populate LOD filter with LOD facet terms

jraddaoui committed:
Fix top-level autocomplete width

jraddaoui committed:
Allow to remove copyright and material filters

- Add settings in migrations and fixtures
- Include settings in default page elements form
- Change filters disposition to allow one or two filters in a row
- Add responsive design for the adv. search partial

jraddaoui committed:
Move all page elements form to settings action

And clean a lot of repeated code

jraddaoui committed:
Remove sfThemePlugin

After 2.x it was only used in the page elements form
and after moving the entire form to the settings action
it's not needed anymore

MikeFE commented on apps/qubit/modules/informationobject/actions/browseAction.class.php:
```diff
@@ -160,76 +624,194 @@ protected function populateFacet($name, $ids)
+      // - Criteria fields
+      // - 'query' param (added to the criteria)
+      // - 'showAdvanced' param (always added)
+      if (in_array($key, $this::$NAMES)
```
too many newlines here I think. Just break on 120 chars maybe.

MikeFE commented on apps/qubit/modules/search/templates/_facet.php:
```diff
@@ -1,76 +1,62 @@
-  <section class="facet">
-<?php endif; ?>
+<?php if ((isset($filters[$facet]) && @count($pager->facets[$facet]['terms']) == 1)
+  || @count($pager->facets[$facet]['terms']) > 1): ?>
```
This probably isn't your code but I think we should be more verbose here, using @ instead of checking isset() a bunch seems like really bad programming practice.

MikeFE commented on apps/qubit/modules/search/templates/_facetLanguage.php:
```diff
@@ -1,51 +1,55 @@
-      <h3><a href="#" aria-expanded="false"><?php echo $label ?></a></h3>
-    <?php endif; ?>
-  </div>
+<?php if (@count($pager->facets[$facet]['terms']) > 2): ?>
```
Same here as above comment

MikeFE commented on apps/qubit/modules/search/templates/_facetLanguage.php:
```diff
@@ -1,51 +1,55 @@
+          <li>
+        <?php endif; ?>
+          <?php echo link_to(
+            __('Unique records') . '<span>, ' . $pager->facets[$facet]['terms']['unique']['count'] . ' ' . __('results') . '</span>',
```
It seems crazy chaining the array [] stuff 4x in a row... but not sure how else we could do it without a bunch of work

MikeFE commented on plugins/sfThemePlugin/config/sfThemePluginConfiguration.class.php:
```diff
@@ -1,21 +0,0 @@
-<?php
```
Please make sure to mention you removed this dead plugin in your commit description

jraddaoui replied:
I've updated the PR description, I'll use it as the commit message.

Thanks @MikeFE

jraddaoui committed:
Cosmetic change

jraddaoui committed:
Improve facets code

- Stop using @ and check existence
- Fix unique languages data and improve conditions

jraddaoui committed:
Fix boolean fields order

jraddaoui committed:
Fix digital object filter tag

jraddaoui committed:
Fix language facet when no results

jraddaoui committed:
Fix search box and first criterion relation

jraddaoui committed:
Don't escape boolean criteria

jraddaoui committed:
Change label

jraddaoui committed:
Check repo and parent type in IO autocomplete

jraddaoui commented:
Merged using '--squash'
